### PR TITLE
Make sure urls have trailing slashes

### PR DESCRIPTION
**Pull request by saltzmanjoelh**
Everything seemed to route through this like you suggested.

### CODE REVIEW

File: Sources/AWSSigner/signer.swift [hunk 1]
```diff
@@ -120,7 +120,12 @@ public struct AWSSigner {
         var date : String { return String(datetime.prefix(8))}
 
         init(url: URL, method: HTTPMethod = .GET, headers: HTTPHeaders = HTTPHeaders(), body: BodyData? = nil, bodyHash: String? = nil, date: String, signer: AWSSigner) {
-            self.url = url
+            if url.absoluteString.hasSuffix("amazonaws.com") {
+                //URL has to have trailing slash
```
Comment (adam-fowler) [inline Sources/AWSSigner/signer.swift:124]: This makes the assumption the URL ends in `amazonaws.com`. Could we not use `url.path == ""`

**Review by adam-fowler — COMMENTED** [review]
Apart from the one issue I think we are good

File: Sources/AWSSigner/signer.swift (hunk 1 above)
Comment (saltzmanjoelh) [reply Sources/AWSSigner/signer.swift:124]: 🤦‍♂️Yeah that makes more sense. Thanks. Changes pushed.

**Review by adam-fowler — APPROVED** [review]
I'm happy with these changes

**Comment by adam-fowler** [comment]
I'll push a release in the next couple of days

**Comment by saltzmanjoelh** [comment]
Cool thanks no rush.

**Pull request merged**